ecl/hql/hqlgram2.cpp

```diff
@@ -1915,7 +1915,7 @@ void HqlGram::addAssignall(IHqlExpression *tgt, IHqlExpression *src, const attri
     IHqlExpression * srcRecord = src->queryRecord();
     IHqlExpression * tgtRecord = tgt->queryRecord();
 
-    if (srcRecord && tgtRecord && recordTypesMatch(srcRecord, tgtRecord) && (tgtOp != no_self) && !haveAssignedToChildren(tgt))
+    if (!lookupCtx.syntaxChecking() && srcRecord && tgtRecord && recordTypesMatch(srcRecord, tgtRecord) && (tgtOp != no_self) && !haveAssignedToChildren(tgt))
```

Comment:
Slightly hesitant that this will kick in if syntax checking - even if simplified expressions not being used.  (Not going through this code could slow parsing down.)
Probably better would be to check ::recordTypesMatch() since this is should just be an optimization.